src/api/Controllers/BoreholeController.cs

```diff
@@ -65,6 +67,36 @@ public async override Task<ActionResult<Borehole>> EditAsync(Borehole entity)
         }
     }
 
+    /// <summary>
+    /// Asynchronously gets all <see cref="Borehole"/> records with optional filtering by ids and pagination.
+    /// </summary>
+    /// <param name="ids">The optional list of borehole ids to filter by.</param>
+    /// <param name="pageNumber">The page number for pagination.</param>
+    /// <param name="pageSize">The page size for pagination.</param>
+    [HttpGet]
+    [Authorize(Policy = PolicyNames.Viewer)]
+    public async Task<ActionResult<PaginatedResponse<Borehole>>> GetAllAsync([FromQuery] IEnumerable<int>? ids = null, [FromQuery] int pageNumber = 1, [FromQuery] int pageSize = 10)
+    {
+        pageSize = pageSize > MaxPageSize ? MaxPageSize : pageSize;
+
+        var skip = (pageNumber - 1) * pageSize;
+
+        var query = GetBoreholesWithIncludes().AsNoTracking();
+
+        if (ids != null && ids.Any())
+        {
+            query = query.Where(borehole => ids.Contains(borehole.Id));
+        }
+
+        var totalCount = await query.CountAsync().ConfigureAwait(false);
+
+        var items = await query.Skip(skip).Take(pageSize).ToListAsync().ConfigureAwait(false);
+
+        var paginatedResponse = new PaginatedResponse<Borehole>(items, totalCount, pageNumber, pageSize, MaxPageSize);
```

Comment:
PP: Zwischen diesen drei Variablendefinitionen braucht es keinen Whitespace, das zieht den Code ziemlich in die Länge. Das Gleiche ist für `skip` und `query` oben auch zu prüfen.